service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/aws/s3storageFolder/ValidateAwsS3StorageFolderCreateStep.java

```diff
@@ -0,0 +1,52 @@
+package bio.terra.workspace.service.resource.controlled.cloud.aws.s3storageFolder;
+
+import bio.terra.common.exception.ConflictException;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.utils.AwsUtils;
+import bio.terra.workspace.service.workspace.AwsCloudContextService;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+
+public class ValidateAwsS3StorageFolderCreateStep implements Step {
+  private final ControlledAwsS3StorageFolderResource resource;
+  private final AwsCloudContextService awsCloudContextService;
+
+  public ValidateAwsS3StorageFolderCreateStep(
+      ControlledAwsS3StorageFolderResource resource,
+      AwsCloudContextService awsCloudContextService) {
+    this.resource = resource;
+    this.awsCloudContextService = awsCloudContextService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    AwsCredentialsProvider credentialsProvider =
+        AwsUtils.createWsmCredentialProvider(
+            awsCloudContextService.getRequiredAuthentication(),
+            awsCloudContextService.discoverEnvironment());
+
+    if (AwsUtils.checkFolderExists(
+        credentialsProvider,
+        Region.of(resource.getRegion()),
+        resource.getBucketName(),
+        resource.getPrefix())) {
+      return new StepResult(
+          StepStatus.STEP_RESULT_FAILURE_FATAL,
+          new ConflictException(
+              String.format(
+                  "Prefix '%s/' already exists in bucket '%s'.",
```

Comment:
Do we expect our users know what we are talking about when we use the terms prefix and bucket here?

Reply:
Noted, changing the text to folder in all error messages which is expected UX nomenclature.

Reply:
handled in https://github.com/DataBiosphere/terra-workspace-manager/pull/1214